### DISCUSSION TITLE
Does Hasura supports oracle DataGuard read replica?

### DISCUSSION THREAD

**Discussion by azran4u** [hasura/graphql-engine#9707] [Question] [answered]
Our primary legacy database is based on Oracle RAC and has a read only replica instance using Oracle DataGuard. We would like Hasura to expose Graphql API based on the replica instance. Oracle DataGuard replica is read only. Does Hasura Enterprise Edition supports such configuration? Do we need another RW database for Hasura's metadata?

**Comment by rsd1122** [comment, marked as answer]
Hi @azran4u 
Hasura Enterprise should work fine with DataGuard as long as you are using Oracle database version 18.0 and higher and Hasura EE is able to connect to it via a JDBC URL. Given that this is a read only instance, mutations (INSERT/UPDATE/DELETE) operations will throw an error as expected. 

And yes, you will need a Postgres database for Hasura's metadata. 

Feel free to get in touch with me (email is in my profile) and we can talk in detail about your use cases.
